src/main/java/org/springframework/cloud/deployer/scheduler/spi/kubernetes/KubernetesScheduler.java

```diff
@@ -81,6 +85,16 @@ public void schedule(ScheduleRequest scheduleRequest) {
 		}
 	}
 
+	public void validateScheduleName(ScheduleRequest request) {
+		if(request.getScheduleName() == null) {
+			throw new IllegalArgumentException("The name for the schedule request is null");
```

Comment:
why throw IllegalArgumentException in both places of this method vs CreateScheduleException? we throw that when the cron expression isnt valid as well

Reply:
Good Catch

Reply:
As it pertains to the cron expression, the exceptions we get back from k8's are pretty good.  
i.e.
```
Failed to create schedule CronJob.batch "bbb" is invalid: spec.schedule: Invalid value: "*/1 * * R C": Failed to parse int from R: strconv.Atoi: parsing "R": invalid syntax
``` 
when I tried to enter a cron expression of `*/1 * * R C`